fdbserver/RestoreMaster.actor.cpp

```diff
@@ -0,0 +1,369 @@
+				wait(tr.commit());
+				wait( watch4RestoreRequest );
+			} else {
+				state Standalone<RangeResultRef> restoreRequestValues = wait(tr.getRange(restoreRequestKeys, CLIENT_KNOBS->TOO_MANY));
+				ASSERT(!restoreRequestValues.more);
+				if(restoreRequestValues.size()) {
+					for ( auto &it : restoreRequestValues ) {
+						restoreRequests.push_back(restoreRequests.arena(), decodeRestoreRequestValue(it.value));
+						printf("Restore Request:%s\n", restoreRequests.back().toString().c_str());
+					}
+				}
+				break;
+			}
+		} catch(Error &e) {
+			wait(tr.onError(e));
+		}
+	}
+
+	return restoreRequests;
+}
+
+// NOTE: This function can now get the backup file descriptors
+ACTOR static Future<Void> _collectBackupFiles(Reference<RestoreMasterData> self, Database cx, RestoreRequest request) {
+	self->initBackupContainer(request.url);
+	state BackupDescription desc = wait(self->bc->describeBackup());
+
+	// TODO: Delete this and see if it works
+	wait(desc.resolveVersionTimes(cx));
+
+	printf("[INFO] Backup Description\n%s", desc.toString().c_str());
+	if(request.targetVersion == invalidVersion && desc.maxRestorableVersion.present())
+		request.targetVersion = desc.maxRestorableVersion.get();
+
+	Optional<RestorableFileSet> restorable = wait(self->bc->getRestoreSet(request.targetVersion));
+
+	if(!restorable.present()) {
+		TraceEvent(SevWarn, "FastRestore").detail("NotRestorable", request.targetVersion);
+		throw restore_missing_data();
+	}
+
+	if (!self->files.empty()) {
+		TraceEvent(SevError, "FastRestore").detail("ClearOldFiles", self->files.size());
+		self->files.clear();
+	}
+
+ 	for(const RangeFile &f : restorable.get().ranges) {
+ 		TraceEvent("FastRestore").detail("RangeFile", f.toString());
+		RestoreFileFR file(f.version, f.fileName, true, f.blockSize, f.fileSize, f.version, f.version);
+ 		self->files.push_back(file);
+ 	}
+ 	for(const LogFile &f : restorable.get().logs) {
+ 		TraceEvent("FastRestore").detail("LogFile", f.toString());
+		RestoreFileFR file(f.beginVersion, f.fileName, false, f.blockSize, f.fileSize, f.endVersion, f.beginVersion);
+		self->files.push_back(file);
+ 	}
+
+	return Void();
+}
+
+ACTOR static Future<Void> _clearDB(Database cx) {
+	wait( runRYWTransaction( cx, [](Reference<ReadYourWritesTransaction> tr) -> Future<Void> {
+			tr->setOption(FDBTransactionOptions::ACCESS_SYSTEM_KEYS);
+			tr->setOption(FDBTransactionOptions::LOCK_AWARE);
+			tr->clear(normalKeys);
+			return Void();
+  		}) );
+
+	return Void();
+}
+
+
+ACTOR static Future<Void> initializeVersionBatch(Reference<RestoreMasterData> self) {
+
+	std::vector<std::pair<UID, RestoreVersionBatchRequest>> requests;
+	for (auto &applier : self->appliersInterf) {
+		requests.push_back( std::make_pair(applier.first, RestoreVersionBatchRequest(self->batchIndex)) );
+	}
+	wait( sendBatchRequests(&RestoreApplierInterface::initVersionBatch, self->appliersInterf, requests) );
+
+	std::vector<std::pair<UID, RestoreVersionBatchRequest>> requests;
```

Comment:
open for ide probably doesn't matter, it's probably just "do a cmake build locally"

Reply:
I don't think I got `, as you have the variety of classic flow mistakes, like redeclaring requests here`.

Because there is a `wait` statement after the first declaration of `requests`, the first `requests` scope ends at the `wait` statement. 
That's why I declare another one after the `wait` statement.

Is there a canonical way to handle this kind of situation?